src/app/mesh/qgsmeshrenderer3daveragingwidget.cpp

```diff
@@ -27,26 +30,31 @@ QgsMeshRenderer3dAveragingWidget::QgsMeshRenderer3dAveragingWidget( QWidget *par
   setupUi( this );
   connect( mAveragingMethodComboBox, qgis::overload<int>::of( &QComboBox::currentIndexChanged ),
            this, &QgsMeshRenderer3dAveragingWidget::onAveragingMethodChanged );
-  QVector<QLineEdit *> widgets;
-  connect( mVerticalLayerIndexSpinBox, qgis::overload<int>::of( &QgsSpinBox::valueChanged ), this, &QgsMeshRenderer3dAveragingWidget::widgetChanged );
+  connect( mVerticalLayerIndexSpinBox, qgis::overload<int>::of( &QgsSpinBox::valueChanged ),
+           this, &QgsMeshRenderer3dAveragingWidget::widgetChanged );
 }
 
 void QgsMeshRenderer3dAveragingWidget::setLayer( QgsMeshLayer *layer )
 {
   mMeshLayer = layer;
 }
 
-QgsMeshRenderer3dAveragingSettings QgsMeshRenderer3dAveragingWidget::settings() const
+QgsMesh3dAveragingMethod *QgsMeshRenderer3dAveragingWidget::averagingMethod() const
```

Comment:
Would be good to return std::unique_ptr here to make it obvious that we are passing the object with ownership...